File: Web/coffee/lib/service/user_service.dart
import 'package:coffee/model/result.dart';
import 'package:coffee/model/user.dart';
import 'package:coffee/util/dio_app.dart';
import 'package:dio/dio.dart';

Future<Result<User>> login(
    String account, String password, int userType) async {
  Response response = await appDio.post("/user/login", queryParameters: {
    "account": account,
    "password": password,
    "type": userType
  });
  Result<User> result =
      Result<User>.fromJson(response.data, (user) => User.fromJson(user));
  return result;
}

Future<Result<User>> addUser(User user) async =>
    await appDio.post("/user/addUser", data: user).then(
        (response) => Result.fromJson(response.data, (s) => User.fromJson(s)));

Future<Result<String>> deleteUser(int userId) async {
  Response response = await appDio
      .post("/user/deleteUser", queryParameters: {"userId": userId});
  Result<String> result =
      Result<String>.fromJson(response.data, (data) => data);
  return result;
}

Future<Result<List<User>>> getUserList() async {
  Response response = await appDio.get("/user/userList");
  Result<List<User>> result = Result.fromJson(response.data,
      (users) => (users as List).map((e) => User.fromJson(e)).toList());
  return result;
}

Future<Result<String>> updateUser(User user) async {
  Response response = await appDio.post("/user/updateUser", data: user);
  Result<String> result = Result.fromJson(response.data, (s) => s);
  return result;
}

Future<Result<User>> getUserById(int userId) async {
  Response response =
      await appDio.get("/user/getUser", queryParameters: {"userId": userId});
  Result<User> result = Result.fromJson(response.data, (s) => User.fromJson(s));
  return result;
}

File: Server/src/main/java/com/drink/coffee/service/UserService.java
package com.drink.coffee.service;

import com.drink.coffee.model.Result;
import com.drink.coffee.pojo.User;

import java.util.List;

public interface UserService {
    /**
     * 用户登录
     *
     * @param account  账号
     * @param password 密码
     * @param userType 用户类型
     * @return 登录的用户
     */
    Result<User> userLogin(String account, String password, int userType);

    /**
     * 用户列表
     *
     * @return 用户列表
     */
    Result<List<User>> userList();

    /**
     * 删除用户
     *
     * @param userId 用户Id
     * @return 删除结果
     */
    Result<String> deleteUser(int userId);

    /**
     * 更新用户信息
     *
     * @param user 更新的用户
     * @return 更新结果
     */
    Result<String> updateUser(User user);

    /**
     * 注册用户
     *
     * @param user 注册的用户
     * @return 注册的用户
     */
    Result<User> addUser(User user);

    /**
     * 获取用户
     *
     * @param userId 用户ID
     * @return 用户
     */
    Result<User> getUserById(int userId);
}
